File: lib/src/main/java/pl/tajchert/sample/InvalidateViewOnUpdate.java
package pl.tajchert.sample;

import android.animation.ValueAnimator;
import android.view.View;
import java.lang.ref.WeakReference;

public class InvalidateViewOnUpdate implements ValueAnimator.AnimatorUpdateListener {
    private final WeakReference<View> viewRef;

    public InvalidateViewOnUpdate(View view) {
        this.viewRef = new WeakReference<>(view);
    }

    @Override
    public void onAnimationUpdate(ValueAnimator valueAnimator) {
        final View view = viewRef.get();

        if (view == null) {
            return;
        }

        view.invalidate();
    }
}

File: lib/src/main/java/pl/tajchert/waitingdots/InvalidateViewOnUpdate.kt
package pl.tajchert.waitingdots

import android.animation.ValueAnimator
import android.view.View
import java.lang.ref.WeakReference

class InvalidateViewOnUpdate(view: View) : ValueAnimator.AnimatorUpdateListener {
  private val viewRef: WeakReference<View> = WeakReference(view)

  override fun onAnimationUpdate(valueAnimator: ValueAnimator) {
    val view = viewRef.get() ?: return

    view.invalidate()
  }
}
